### PR TITLE
Add Docc Support and Some Preliminary Articles

### PR DESCRIPTION
**Pull request by CodaFi**
This provides a model for what integrating Docc documentation would look like for the Swift driver. A few articles are included to get a feel for the basics of what we can provide here.

### CODE REVIEW

File: Sources/SwiftDriver/SwiftDriver.docc/IncrementalBuilds.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
+# Incremental Builds
```
Comment (davidungar) [inline Sources/SwiftDriver/SwiftDriver.docc/IncrementalBuilds.md:1]: This whole file is great stuff! Thanks!

**Review by davidungar — APPROVED** [review]
Thanks for doing this! LGTM

File: Sources/SwiftDriver/SwiftDriver.docc/IncrementalBuilds.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+The build begins by examining "priors" - data left behind by the last compilation
+session. If no data is found, the Driver considers the incremental build to be
+a lost cause and schedules a full rebuild in order to gather `.swiftdeps` files
+it can use to reconstruct this prior data.
+
+Assuming priors are present, they are deserialized and any dependency 
+information they contain is integrated into the driver's dependency graph. Next,
+the modification time of any files is examined. If these modification times do
+not match the driver's last expected modification time, those files are 
+immediately scheduled for rebuilding. Using the information from the prior build,
```
[the remark this answers is not in the document]
Comment (CodaFi) [reply Sources/SwiftDriver/SwiftDriver.docc/IncrementalBuilds.md:119]: We do re-compute the interface hash in the frontends, but the driver does not do any additional stuff with it besides potentially use it in second-wave scheduling when that hash disagrees with a prior.

File: Sources/SwiftDriver/SwiftDriver.docc/IncrementalBuilds.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+The Swift compilation model includes one more wrinkle: the compilation of one
+file may introduce dependencies on another file that needs to rebuild afterward.
+Therefore, the incremental build state can be continuously queried to both
+integrate new dependencies and discover additional compilation jobs to run. This
+implies that the incremental build must be run *to a fixpoint* in order for
+it to complete.
+
+## Communicating Dependencies
+
+The Swift driver includes built-in facilities for reading incremental dependnecy
```
Comment (tschuett) [inline Sources/SwiftDriver/SwiftDriver.docc/IncrementalBuilds.md:30]:
```suggestion
The Swift driver includes built-in facilities for reading incremental dependency
```

File: Sources/SwiftDriver/SwiftDriver.docc/SwiftDriver.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+## Overview
+
+The `SwiftDriver` framework coordinates the compilation of Swift source code
+into various compiled results: executables, libraries, object files, Swift 
+modules and interfaces, etc. It is the program one invokes from the command line
+to build Swift code (i.e., swift or swiftc) and is often invoked on the 
+developer's behalf by a build system such as the 
+[Swift Package Manager](https://github.com/apple/swift-package-manager)
+or Xcode's build system.
+
```
Comment (tschuett) [inline Sources/SwiftDriver/SwiftDriver.docc/SwiftDriver.md:14]: It is not obvious that the driver is a large library and a tiny `main.swift` and the library is designed to be integrated into other build systems.

File: Sources/SwiftDriver/SwiftDriver.docc/IncrementalBuilds.md (hunk 3 above)
Comment (CodaFi) [reply Sources/SwiftDriver/SwiftDriver.docc/IncrementalBuilds.md]: Thanks.

File: Sources/SwiftDriver/SwiftDriver.docc/SwiftDriver.md (hunk 4 above)
Comment (CodaFi) [reply Sources/SwiftDriver/SwiftDriver.docc/SwiftDriver.md:14]: I will change the wording here to be more library-oriented. I figure once we have dedicated docs for `swift-driver` and `swift-help`, etc that we can link out to those here.

**Comment by tschuett** [comment]
I believe in some cases a which has a leading , and in some cases it is missing.

**Review by nkcsgexi — APPROVED** [review]

**Comment by CodaFi** [comment]
@swift-ci test

**Comment by CodaFi** [comment]
⛵

**Pull request merged**